From f40dbfdcee004f078216b00b1ce46744a8664205 Mon Sep 17 00:00:00 2001
From: Leonidas Spyropoulos <artafinde@gmail.com>
Date: Tue, 14 Jun 2022 18:00:54 +0100
Subject: [PATCH] Add support for reproducible builds

As per gradle [docs] add support to remove timestamps and package with same order which is required from
[reproducible] builds

[docs]: https://docs.gradle.org/current/userguide/working_with_files.html#sec:archives
[reproducible]: https://reproducible-builds.org/

Signed-off-by: Leonidas Spyropoulos <artafinde@archlinux.com>
---
 build.gradle | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/build.gradle b/build.gradle
index 60b11af..9a898d3 100644
--- a/build.gradle
+++ b/build.gradle
@@ -88,6 +88,10 @@ allprojects {
         compileKotlin.kotlinOptions.jvmTarget = compileTestKotlin.kotlinOptions.jvmTarget = JavaVersion.VERSION_11
         compileKotlin.dependsOn ktlint
     }
+    tasks.withType(AbstractArchiveTask).configureEach {
+        preserveFileTimestamps = false
+        reproducibleFileOrder = true
+    }
 }
 
 evaluationDependsOnChildren()
-- 
2.36.1

